Assets/Scripts/Battle/UIController.cs

```diff
@@ -978,8 +978,14 @@ public enum UIState {
                         }
                         case 1: {
                             if (!GlobalControls.retroMode) {
-                                if ((EnemyEncounter.script.GetVar("fleesuccess").Type != DataType.Boolean && (Math.RandomRange(0, 9) + encounter.turnCount) > 4)
-                                 || EnemyEncounter.script.GetVar("fleesuccess").Boolean)
+                                bool fleeSuccess = (
+                                    EnemyEncounter.script.GetVar("fleesuccess").Type != DataType.Boolean
+                                    && (Math.RandomRange(0, 9) + encounter.turnCount) > 4
+                                ) || EnemyEncounter.script.GetVar("fleesuccess").Boolean;
+
+                                encounter.CallOnSelfOrChildren("HandleFlee", new DynValue[] {DynValue.NewBoolean(fleeSuccess)});
+
+                                if (fleeSuccess)
```

Comment:
The way you implemented HandleFlee() makes it so that anything the user tries to do regarding states is disregarded, be careful!
If for example I want to implement a dialogue with HandleFlee() like I would with OnDeath(), the fleeing sequence will still be started, which is bad!
Might as well prevent the engine's fleeing sequence to run if HandleFlee() exists and implement the engine function Flee(bool success) that initiates the engine's fleeing sequence!